src/ImageSharp/Metadata/Profiles/ICC/MultiProcessElements/IccMatrixProcessElement.cs

```diff
@@ -52,13 +52,17 @@ public override bool Equals(IccMultiProcessElement other)
 
         /// <inheritdoc />
         public bool Equals(IccMatrixProcessElement other)
-        {
-            return this.Equals((IccMultiProcessElement)other);
-        }
+            => this.Equals((IccMultiProcessElement)other);
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+            => this.Equals(obj as IccMatrixProcessElement);
```

Comment:
We could also directly do `IccMultiProcessElement` here?

Reply:
The cast is the correct type no?

Reply:
It is correct but the Equals that you are calling does a cast to `IccMultiProcessElement` so you could decide to directly cast instead. Same goes for my other comment.

Reply:
Ah right, that depends on an implementation detail of the explicit equality comparer method (which I want to look into. Why are there separate types that do not vary by properties only property values?)

For safety I would always attempt a cast to the concrete type.